packages/story-editor/src/app/taxonomy/taxonomyProvider.tsx

```diff
@@ -0,0 +1,285 @@
+      return;
+    }
+
+    void (async () => {
+      try {
+        const result = await getTaxonomies();
+        setTaxonomies(result);
+      } catch (e) {
+        if (e instanceof Error) {
+          // eslint-disable-next-line no-console -- Log error
+          console.error(e.message);
+        }
+      } finally {
+        setHasTaxonomies(true);
+      }
+    })();
+  }, [hasTaxonomies, getTaxonomies]);
+
+  const setTerms = useCallback(
+    ({ newTerms }: setTermsProps) => {
+      if (updateStory) {
+        const properties = (story: Story) => {
+          const currentTerms = story?.terms || [];
+          const newAssignedTerms = [...new Set([...currentTerms, ...newTerms])];
+          return {
+            ...story,
+            terms: newAssignedTerms,
+          };
+        };
+        updateStory({ properties } as UpdateStoryProps);
+      }
+    },
+    [updateStory]
+  );
+
+  const removeTerms = useCallback(
+    ({ removeTerms }: removeTermsProps) => {
+      if (updateStory) {
+        const removeTermsID = removeTerms.map(({ id }) => id);
+        const properties = (story: Story) => {
+          const currentTerms = story?.terms || [];
+          const newAssignedTerms = currentTerms.filter(
+            ({ id }) => !removeTermsID.includes(id)
+          );
+          return {
+            ...story,
+            terms: newAssignedTerms,
+          };
+        };
+        updateStory({ properties } as UpdateStoryProps);
+      }
+    },
+    [updateStory]
+  );
+
+  const addTermToSelection = useCallback(
+    ({ newTerms }: addTermToSelectionProps) => setTerms({ newTerms }),
+    [setTerms]
+  );
+
+  const addSearchResultsToCache = useCallback(
+    async ({
+      taxonomy,
+      args,
+      addNameToSelection = false,
+    }: addSearchResultsToCacheProps) => {
+      let termResults: Term[] = [];
+      const termsEndpoint = taxonomy?.restPath;
+      if (!termsEndpoint || !getTaxonomyTerm) {
+        return [];
+      }
+      try {
+        termResults = await getTaxonomyTerm(termsEndpoint, {
+          search: args?.search,
+          per_page: args?.per_page,
+        });
+      } catch (e) {
+        // Log error
+        if (e instanceof Error) {
+          // eslint-disable-next-line no-console -- We want to surface this error.
+          console.error(e.message);
+        }
+      }
+
+      // Avoid update if we're not actually adding any terms here
+      if (termResults.length < 1) {
+        return termResults;
+      }
+
+      setTermCache((cache: Term[]) => {
+        return cache ? [...cache, ...termResults] : termResults;
+      });
+
+      if (addNameToSelection && args.search) {
+        const selectedTermSlug: TermSlug = cleanForSlug(args.search);
+        const selectedTerm = termResults.find(
+          (term: Term) => term.slug === selectedTermSlug
+        );
+
+        if (selectedTerm) {
+          addTermToSelection({ newTerms: [selectedTerm] });
```

Comment:
As per my other suggestions, this can be simplified to

```suggestion
          addTermToSelection({ newTerms: [selectedTerm] });
```